src/main/java/ro/utcn/sd/agui/a1/service/AnswerVoteManagementService.java

```diff
@@ -0,0 +1,123 @@
+import ro.utcn.sd.agui.a1.entity.*;
+import ro.utcn.sd.agui.a1.persistence.*;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class AnswerVoteManagementService {
+
+    private final RepositoryFactory repositoryFactory;
+
+    @Transactional
+    public List<AnswerVote> listAllAnswerVotes() {
+        return repositoryFactory.createAnswerVoteRepository().findAll();
+    }
+
+    @Transactional
+    public int addAnswerVote(int answerId, int userId, Boolean type) {
+
+        AnswerVoteRepository answerVoteRepository = repositoryFactory.createAnswerVoteRepository();
+        AnswerRepository answerRepository = repositoryFactory.createAnswerRepository();
+        UserRepository userRepository = repositoryFactory.createUserRepository();
+
+        Optional<Answer> foundAnswer = answerRepository.findById(answerId);
+        Optional<User> foundUser = userRepository.findById(userId);
+
+        if (foundAnswer.isPresent()) {
+            if (userId != foundAnswer.get().getUserId()) {
+                //Optional<AnswerVote> foundVote = answerVoteRepository.findByUserAndQuestion(foundUser.get(), foundQuestion.get());
+                //if(foundVote.isPresent()){
+                //    return -3; //the vote is already present
+                //}
+                //else{
+                AnswerVote newAnswerVote = new AnswerVote(null, answerId, userId, type);
+                return answerVoteRepository.save(newAnswerVote).getVoteId();
+                // }
+            } else {
+                return -2; //cannot vote own question
+            }
+        } else {
+            return -1; //no question
+        }
+    }
+
+    @Transactional
+    public int deleteAnswerVote(int voteId, int userId) {
+
+        AnswerVoteRepository answerVoteRepository = repositoryFactory.createAnswerVoteRepository();
+        Optional<AnswerVote> foundVote = answerVoteRepository.findById(voteId);
+
+        if (foundVote.isPresent()) {
+            if (foundVote.get().getUserId() == userId) {
+                answerVoteRepository.remove(foundVote.get());
+                return voteId;
+            } else {
+                return -2; //the vote is not of the current user
+            }
+        } else {
+            return -1; //vote not found
+        }
+    }
+
+    @Transactional
+    public int updateAnswerVote(int voteId, int userId, Boolean type) {
+
+        AnswerVoteRepository answerVoteRepository = repositoryFactory.createAnswerVoteRepository();
+        Optional<AnswerVote> foundVote = answerVoteRepository.findById(voteId);
+
+        if (foundVote.isPresent()) {
+            if (foundVote.get().getUserId() == userId) {
+                AnswerVote updatedVote = foundVote.get();
+                updatedVote.setType(type);
+                answerVoteRepository.save(updatedVote);
+                return voteId;
+            } else {
+                return -2; //the vote is not of the current user
```

Comment:
See, you have to use comments to explain what is happening here. It would be a lot clearer from the code itself if you would have exceptions / Optional instead.